lib/cli/handle-error.js

```diff
@@ -141,7 +141,10 @@ module.exports = async (exception, options = {}) => {
   consoleLog(' ');
 
   // If `hasTelemetryBeenReported` is not defined, we don't want to publish telemetry
-  if (!isTelemetryDisabled && hasTelemetryBeenReported === false) {
+
+  const isForInvalidCommand =
+    exception.code === 'COMMAND_NOT_FOUND' || exception.code === 'UNRECOGNIZED_CLI_COMMAND';
+  if (!isForInvalidCommand && !isTelemetryDisabled && hasTelemetryBeenReported === false) {
```

Comment:
More bulletproof will be simply to avoid storing telemetry if `resolveInput()` doesn't resolve `commandSchema`.

Technically we can introduce other kind of errors prior commands validation and then again this place will leak again

Reply:
That's a great point - changed